File: dotnet/src/Azure/Mgmt/Microsoft.Compute/2016_06_30/Operations/VirtualMachinesOperations.cs
using Azure.Mgmt.Models;
using Azure.Mgmt.Compute._2016_06_30.Models;
using System.Collections.Generic;

namespace Azure.Mgmt.Compute._2016_06_30.Operations {
    public class VirtualMachinesOperations {
        internal VirtualMachinesOperations(AzureClient client){

        }

        public IVirtualMachine Create(string name, string location, IDictionary<string, string> tags = null, string licenseType = null, string vmId = null, IPlan plan = null){
            return new VirtualMachine(name, location, tags, licenseType, vmId, plan);
        }
    }
}

File: ruby/src/azure/mgmt/microsoft_compute/2016_01_31/operations/virtual_machines_operations.rb
require_relative "../models/virtual_machine"

module Azure
  module Mgmt
    module Compute
      module Api_2016_01_31
        module Operations
          class VirtualMachinesOperations
            def initialize(client)
            end

            def create(name:, location:, tags: nil, vm_id: nil, disk:)
              Azure::Mgmt::Compute::Api_2016_01_31::Models::VirtualMachine.new(name: name,
                                                                               location: location,
                                                                               tags: tags,
                                                                               vm_id: vm_id)
            end
          end
        end
      end
    end
  end
end